qtfred/src/CheckedDataListModel.h

```diff
@@ -99,15 +101,20 @@ class CheckedDataListModel : public QAbstractListModel
 		else if (items[srow]._color != Qt::color0 && role == Qt::ForegroundRole)
 			return QBrush(items[srow]._color);
 		else
-			return QVariant();
+			return {};
 	}
 
-	inline T& internalData(const QModelIndex &index) {
-		return items[static_cast<size_t>(index.row())].internalData();
+	inline const T* internalData(const QModelIndex &index) const {
+		if (! index.isValid()) return nullptr;
+		size_t row{static_cast<size_t>(index.row())};
+		if (row >= items.size()) return nullptr;
+		return &items[row].internalData();
 	}
 
-	inline const T& internalDataConst(const QModelIndex &index) const {
-		return items[static_cast<size_t>(index.row())].internalData();
+	inline T* internalData(const QModelIndex &index) {
+		return const_cast<T*>(
```

Comment:
This avoids duplicating the function above: It calls the const version of the method (above), then casts away the const-ness of the return pointer. This is safe, because `this`, and the member returned, are non-const here. [SO](https://stackoverflow.com/questions/123758/how-do-i-remove-code-duplication-between-similar-const-and-non-const-member-funcl)